numpy/lib/tests/test_financial.py

```diff
@@ -344,3 +346,25 @@ def test_broadcast_decimal(self):
                                     Decimal('0'), [Decimal('0'), Decimal('0'), Decimal('1'), 'end', 'begin']),
                             [Decimal('-74.998201'), Decimal('-75.62318601'), Decimal('-75.62318601'),
                              Decimal('-76.88882405'), Decimal('-76.88882405')], 4)
+        
+    # Test for checking inputs whose output is NaN
+    @pytest.mark.parametrize('number_type', [Decimal, float])
+    def test_rate_nan(self, number_type):
```

Comment:
nit: test descriptions should go as `""" docstrings """` inside the function, not comments before